bqskit/ir/circuit.py

```diff
@@ -2752,108 +2761,8 @@ def instantiate(
 
         instantiater = cast(Instantiater, instantiater)
 
-        # Check Target
-        try:
```

Comment:
Removing this casting is a big change, that breaks the API, are you sure you want to do that?

Reply:
I think the casting there was redundant as it was also cast inside Instantiater as well.